core/src/main/java/com/alexafit/core/di/CoreModule.kt

```diff
@@ -0,0 +1,57 @@
+package com.alexafit.core.di
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.SharedPreferencesMigration
+import androidx.datastore.preferences.core.PreferenceDataStoreFactory
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.preferencesDataStore
+import androidx.datastore.preferences.preferencesDataStoreFile
+import com.alexafit.core.BuildConfig
+import com.alexafit.core.local.datastore.DataStorePreferenceStorageImpl
+import com.alexafit.core.local.datastore.PreferenceStorage
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.qualifiers.ApplicationContext
+import dagger.hilt.components.SingletonComponent
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object CoreModule {
+    @Singleton
+    @Provides
+    fun provideDataStore(
+        @ApplicationContext context: Context
+    ): DataStore<Preferences> {
+        val datastore = PreferenceDataStoreFactory.create(
+            migrations = listOf(
+                SharedPreferencesMigration(context, BuildConfig.DATASTORE_PREFERENCE_NAME)
+            ),
+            scope = CoroutineScope(Dispatchers.Default)
```

Comment:
Is there a reason you want to use the default dispatcher for IO stuff? I assume DataStore will switch to IO anyways, but why provide this scope in the first place?
Also, why do you provide a migration here? Migrations are needed if some users already have your app installed which uses shared preferences and you then want to change it to DataStore. Then you'll need to migrate the old shared preferences file to DataStore so it has the same contents and no data is lost. But since this is the very initial release of Tasky, there's no need for migrations